// reve-sdk/dist/types/index.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ReveAIError = exports.ReveAIErrorType = exports.GenerationStatus = void 0;
/**
 * Status of a generation task
 */
var GenerationStatus;
(function (GenerationStatus) {
    GenerationStatus["PENDING"] = "pending";
    GenerationStatus["PROCESSING"] = "processing";
    GenerationStatus["COMPLETED"] = "completed";
    GenerationStatus["FAILED"] = "failed";
})(GenerationStatus || (exports.GenerationStatus = GenerationStatus = {}));
/**
 * Error types that can occur during SDK operations
 */
var ReveAIErrorType;
(function (ReveAIErrorType) {
    ReveAIErrorType["AUTHENTICATION_ERROR"] = "authentication_error";
    ReveAIErrorType["API_ERROR"] = "api_error";
    ReveAIErrorType["REQUEST_ERROR"] = "request_error";
    ReveAIErrorType["TIMEOUT_ERROR"] = "timeout_error";
    ReveAIErrorType["GENERATION_ERROR"] = "generation_error";
    ReveAIErrorType["POLLING_ERROR"] = "polling_error";
    ReveAIErrorType["UNEXPECTED_RESPONSE"] = "unexpected_response";
    ReveAIErrorType["UNKNOWN_ERROR"] = "unknown_error";
})(ReveAIErrorType || (exports.ReveAIErrorType = ReveAIErrorType = {}));
/**
 * Custom error class for SDK errors
 */
class ReveAIError extends Error {
    constructor(message, type = ReveAIErrorType.UNKNOWN_ERROR, statusCode) {
        super(message);
        this.name = 'ReveAIError';
        this.type = type;
        this.statusCode = statusCode;
    }
}
exports.ReveAIError = ReveAIError;
